src/driver.jl

```diff
@@ -183,6 +191,77 @@ const __llvm_initialized = Ref(false)
     entry = finish_module!(job, ir, entry)
 
     # deferred code generation
+    if haskey(functions(ir), "gpuc.lookup")
```

Comment:
Seeing how this doesn't set `has_deferred_jobs`, you probably need to update the use of that conditional during optimization below (adding some additional passes to clean up deferred invokes).